src/azul/openapi.py

```diff
@@ -125,25 +142,70 @@ def clean_specs(specs):
 
 
 def get_app_specs(app):
+    """
+    Extract OpenAPI specs from a Chalice app object
+    """
     path_specs = {}
     method_specs = {}
     for entries in app.routes.values():
         func = next(iter(entries.values())).view_function
-        if hasattr(func, 'path_specs'):
+        try:
             for path, spec in func.path_specs.items():
                 if path in path_specs:
                     assert spec == path_specs[path], f'Path spec for {spec} already exists'
-            path_specs.update(func.path_specs)
-        else:
+        except AttributeError:
             # TODO (jesse): warn if route is undocumented
```

Comment:
Is this hard to do? When do you plan to do this?

Reply:
This can't be done until all of the routes are documented.

Reply:
You can't *warn* unless all routes are documented?

Reply:
I changed "warn" to "raise", because that was what I intended, and it cannot be done until all of the routes are documented.

Reply:
I don't think we want to _actually_ make a warning log here because then it will show up in Airbrake, etc. which I believe we don't want.